main.rb

```diff
@@ -0,0 +1,208 @@
+require 'rubygems'
+require 'nokogiri'
+require 'open-uri'
+require 'watir'
+require 'selenium-webdriver'
+require 'json'
+require 'crack'
+require 'headless'
+require 'colorize'
+
+class Accounts
+  attr_accessor :name, :balance, :currency, :nature
+
+  def initialize(name, balance, currency, nature)
+    @name = name.chomp(' >') # оставил потому что это мой ключ для селекта и сравниваю с элементом у которого нет знака >
+    @balance = balance
+    @currency = currency
+    @nature = nature
+  end
+
+  def to_hash
+    {
+    name: @name,
+    balance: @balance,
+    currency: @currency,
+    nature: @nature
+    }
+  end
+end
+
+class Transactions
+  attr_accessor :date, :description, :amount, :currency, :account_name
+
+  def initialize(date, description, amount, currency, account_name)
+    @date = date
+    @description = description
+    @amount = amount
+    @currency = currency
+    @account_name = account_name
+  end
+
+  def to_hash
+    {
+    date: @date,
+    description: @description,
+    amount: @amount,
+    currency: @currency,
+    account_name: @account_name
+    }
+  end
+end
+
+$URL = 'https://my.fibank.bg/oauth2-server/login?client_id=E_BANK'
+
+class Fibank
+  def my_accounts
+    Selenium::WebDriver::Chrome.driver_path="C:/chromedriver_win32/chromedriver1.exe"
+    # чтобы наблюдать за запуском браузера и нажатиями в браузере отключить параметр headless
+    $browser = Watir::Browser.new :chrome, headless: true
+    $browser.goto $URL
+
+    l = $browser.link text: 'English'
+    l.exists?
+    l.click
+
+    l = $browser.link text: 'Demo'
+    l.exists?
+    l.click
+    sleep 2
+
+    l = $browser.link text: 'See all'
+    l.exists?
+    l.click
+    sleep 2
+
+    l = $browser.link text: 'Balance'
+    l.exists?
+    l.click
+    sleep 1
+    # счетчик количества аккаунтов
+    $exit_condition = $browser.span(css: "#main_grid > div > div:nth-child(2) > div:nth-child(2) > div > div > div > span").text.to_i
+
+    def get_array(name_array, selector)
+      var = 0
+      while var < $exit_condition
+        name_array[var] = $browser.element(index: var, css: selector).text.strip
+        var += 1
+      end
+    end
+
+    # получаем массивы данных
+    get_array(name = [], "#step2 > div > div.blue-bg.h-68 > a")
+    get_array(currency = [], "#step2 > div > div:nth-child(3) > div:nth-child(5) > span")
+    get_array(cur_balance = [], "#step2 > div > div:nth-child(3) > div:nth-child(2) > span")
+
+    i = 0
+    $account_json = []
+    while i < $exit_condition
+      user = Accounts.new(name[i], cur_balance[i].to_f, currency[i], 'checking')
+      hash = {}
+      user.instance_variables.each { |var| hash[var.to_s.delete("@")] = user.instance_variable_get(var) }
+      $account_json[i] = hash
+    #  $account_json[i] = to_hash() остановился здесь не видит метод to_hash
+      i += 1
+    end
+  end
+
+  def my_transactions
+    Selenium::WebDriver::Chrome.driver_path="C:/chromedriver_win32/chromedriver1.exe"

```

Comment:
1) No need to initiate a browser 2 times.  
2) Create a method `connect` and move this logic.
3) You can use the `wait_until_present` method.
4) Also, no need to check for `exists?` all times.

```ruby
def connect
    browser = Watir::Browser.new :chrome, headless: true
    browser.goto URL
    browser.link(text: 'English').wait_until_present.click
end
```